src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/cell/FileCellEditorImpl.java

```diff
@@ -1,27 +1,36 @@
 package org.sagebionetworks.web.client.widget.table.v2.results.cell;
 
 import org.sagebionetworks.web.client.StringUtils;
-import org.sagebionetworks.web.client.widget.upload.FileInputWidget;
-import org.sagebionetworks.web.client.widget.upload.FileMetadata;
-import org.sagebionetworks.web.client.widget.upload.FileUploadHandler;
+import org.sagebionetworks.web.client.utils.CallbackP;
+import org.sagebionetworks.web.client.widget.upload.FileHandleUploadWidget;
+import org.sagebionetworks.web.client.widget.upload.FileUpload;
 
+import com.google.gwt.core.shared.GWT;
 import com.google.gwt.event.dom.client.KeyDownHandler;
 import com.google.gwt.event.shared.GwtEvent;
 import com.google.gwt.event.shared.HandlerRegistration;
 import com.google.gwt.user.client.ui.Widget;
 import com.google.inject.Inject;
 
-public class FileCellEditorImpl implements FileCellEditor, FileCellEditorView.Presenter, FileUploadHandler{
+public class FileCellEditorImpl implements FileCellEditor, FileCellEditorView.Presenter {
 	
 	public static final String MUST_BE_A_FILE_ID_NUMBER = "Must be a File ID number";
 	public static final String PLEASE_SELECT_A_FILE_TO_UPLOAD = "Please select a file to upload.";
 	FileCellEditorView view;
-	FileInputWidget fileInputWidget;
+	FileHandleUploadWidget fileInputWidget;
 	
 	@Inject
-	public FileCellEditorImpl(FileCellEditorView view, FileInputWidget fileInputWidget){
+	public FileCellEditorImpl(final FileCellEditorView view, FileHandleUploadWidget fileInputWidget){
 		this.view = view;
 		this.fileInputWidget = fileInputWidget;
+		fileInputWidget.configure("Browse...", new CallbackP<FileUpload>() {

```

Comment:
Extract "Browse..." to webconstants (it should be the standard file input button text)